components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala

```diff
@@ -96,29 +143,31 @@ case class LiteralRegExpParameterValidator(pattern: String, message: String, des
   lazy val regexpPattern: Pattern = Pattern.compile(pattern)
 
   // Blank value should be not validate - we want to chain validators
-  override def isValid(paramName: String, value: String, label: Option[String])(
+  override def isValid(paramName: String, value: Any, label: Option[String])(
       implicit nodeId: NodeId
   ): Validated[PartSubGraphCompilationError, Unit] = {
     def toResult(validated: Boolean) =
       if (validated) valid(()) else invalid(MismatchParameter(message, description, paramName, nodeId.id))
-
-    if (value.matches("^'.*'$")) { // workaround for SpEL comprehension
-      val trimmedSPeLValue = value.replaceAll("^'|'$", "")
-      toResult(regexpPattern.matcher(trimmedSPeLValue).matches())
-    } else {
-      toResult(StringUtils.isBlank(value) || regexpPattern.matcher(value).matches())
+    value match {
+      case s: String => toResult(StringUtils.isBlank(s) || regexpPattern.matcher(s).matches())
```

Comment:
this `StringUtils.isBlank` case was to handle `null` not the blank string. We don't want to always allow blanks - no matter which regex you will have. Please add a test case for that if it was missing :)